api-inference-community/docker_images/timm/app/pipelines/image_classification.py

```diff
@@ -4,16 +4,28 @@
 import torch
 from app.pipelines import Pipeline
 from PIL import Image
+from timm.data import resolve_data_config
+from timm.data.transforms_factory import create_transform
+from timm.models.hub import load_model_config_from_hf, load_state_dict_from_hf
 
 
 class ImageClassificationPipeline(Pipeline):
     def __init__(self, model_id: str):
-        if "resnet50d" in model_id:
-            self.model = timm.create_model("resnet50d", pretrained=True).eval()
-        elif "dpn92" in model_id:
-            self.model = timm.create_model("dpn92", pretrained=True).eval()
-        else:
-            raise EnvironmentError("Cannot detect the correct arch for this model")
+        self.hf_cfg, self.arch = load_model_config_from_hf(model_id)
+        self.model = timm.create_model(f"hf_hub:{model_id}")
+        state_dict = load_state_dict_from_hf(model_id)
+        self.model.load_state_dict(state_dict, strict=True)
+        self.model.eval()
+
+        self.transform = create_transform(**resolve_data_config({}, model=self.model))
+        self.top_k = min(self.model.num_classes, 5)
+
+        self.labels = self.hf_cfg.get("labels", None)
+        if self.labels is None:
+            if self.model.num_classes == 1000:
```

Comment:
That seems overly specific. Is there no way to fix the `labels` instead  (For the models that need it)?

Reply:
I suppose folks pushing timm models could be instructed to include the labels when they push. that way we wouldn't have to do any of this hard-coded business (which is horrifying to me as well).

In fact, let's do that. If I don't do it for this PR (running out of time on a friday), I'll issue another one early next week that does. 😄

Reply:
Agree with @Narsil, the 1k hardcoding is not great. I think this responsibility should be for the users.